sdk/dotnet/Pulumi.Automation/Serialization/CancelEventModel.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using Pulumi.Automation.Serialization.Json;
+using Pulumi.Automation.Events;
+
+// NOTE: The classes in this file are intended to align with the serialized
+// JSON types defined and versioned in sdk/go/common/apitype/events.go
+namespace Pulumi.Automation.Serialization
+{
+    internal class CancelEventModel : IJsonModel<CancelEvent>
+    {
+        public CancelEvent Convert() =>
+            new CancelEvent();
```

Comment:
You don't need this model, you can just deserialize directly to `CancelEvent` if it has a parameterless constructor. These models are only necessary to get around System.Text.Json not supporting deserialization of parameterized constructors at this version